bgpd/bgp_fsm.c

```diff
@@ -2196,7 +2196,8 @@ void bgp_fsm_nht_update(struct peer *peer, bool has_valid_nexthops)
 	case OpenConfirm:
 	case Established:
 		if (!has_valid_nexthops
-		    && (peer->gtsm_hops == BGP_GTSM_HOPS_CONNECTED))
+		    && (peer->gtsm_hops == BGP_GTSM_HOPS_CONNECTED
+			|| peer->bgp->fast_convergence))
```

Comment:
Can't this be valid for EBGP multihop sessions?

Reply:
This change is a generic change and applies to both IBGP and EBGP. But today EBGP sessions are immediately brought down. Hence this change will help IBGP to behave in the similar way.

Reply:
Even EBGP multihop?

Reply:
yes. When this new CLI is configured, even EBGP multi-hop sessions will be brought down immediately when nh becomes unreachable.
I will probably add this statement explicitly in my commit message and documentation.

Reply:
Then yes, documentation adjustment is needed to mention not only iBGP peers.

Reply:
I have updated it accordingly.